controller/lib/hal/hal.h

```diff
@@ -145,9 +153,14 @@ class HalApi {
   void test_setAnalogPin(AnalogPinId pin, int value);
 #endif
 
-  void analogWrite(PwmPinId pin, int value);
-
+  // TODO(jlebar): Make digital pin number strongly typed?  It's slightly
+  // tricky because the digital pin numbers are a superset of PWM pin numbers,
+  // and C++ doesn't support enum inheritance.
```

Comment:
How about we set all pins to the correct mode for our device during HAL initialization? Though that might be best done in the same PR that raises the level of abstraction from "pin 5" to "pressure pin".

Reply:
Great idea.  Making this happen is a bigger change, so I'll send it in another PR if that's OK.